kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/filter/RequestResponseMarkingFilter.java

```diff
@@ -51,10 +51,13 @@ public RequestResponseMarkingFilter(FilterFactoryContext constructionContext, Re
 
     @Override
     public CompletionStage<RequestFilterResult> onRequest(ApiKeys apiKey, RequestHeaderData header, ApiMessage body, FilterContext context) {
+        if (!constructionContext.filterDispatchExecutor().isInFilterDispatchThread()) {
+            return CompletableFuture.failedFuture(new RuntimeException("onRequest method was not invoked by dispatch thread! "
+                    + "The framework is breaking a threading guarantee"));
+        }
```

Comment:
This should probably be a bit more explicit that its a bug in Kroxylicious not user code and thus they should raise an issue. Possibly an `IllegalStateException`?